### DISCUSSION TITLE
Is it possible to warn or raise error when any function which makes use of `getCurrentInstance` is used after awaited expression in `async setup()`?

### DISCUSSION THREAD

**Discussion by mukundshah** [vuejs/language-tools#3610] [General]
I got assigned on an old project, at my workplace. It was migrated to Nuxt 6 months ago. 

Every vue file is using,
```ts
export default {
  async setup() {
     // any awaited function
     // any function which makes use of getCurrentInstance() directly of indirectly, usually through useNuxtApp
  }
}
```
This shouldn't have worked at all. But it was working until the day I updated all the packages and the lock file. This shouldn't have worked in the first place, unless there was some package inside the lockfile that particularly allowed this. 

The version of vue, vue sfc compiler and nuxt were, before I did the update
vue: `3.3.4`
@vue/compiler-sfc: `3.3.4`
nuxt: `3.6.0`

At these version, they shouldn't have worked, or am I missing something?
Beside this, is it possible to cover/prevent situations like this through volar, or maybe eslint?

**Comment by so1ve** [comment]
I think this is out-of-scope; eslint-plugin-vue should create a rule for it.

**Reply by mukundshah** [reply]
I will link this there then. I was little confused regarding which layer would be able to cover this.

Thank you!

**Reply by so1ve** [reply]
Using `getCurrentInstance` after `await` will cause unexpected behavior in vue, however it's totally valid syntax in both Vue(you will get `null` tho) and JavaScript. Since this behavior is not what you want, so this feature should be covered in static analysis tools to avoid hidden errors.
